presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveType.java

```diff
@@ -199,11 +200,27 @@ private static HiveType toHiveType(TypeInfo typeInfo)
         return new HiveType(typeInfo);
     }
 
-    public static HiveType toHiveType(TypeTranslator typeTranslator, Type type)
+    public static HiveType toHiveType(
+            TypeTranslator typeTranslator,
+            Type type)
+    {
+        return toHiveType(typeTranslator, type, Optional.empty());
+    }
+
+    public static HiveType toHiveType(
+            TypeTranslator typeTranslator,
+            Type type,
+            Optional<HiveType> defaultHiveType)
     {
         requireNonNull(typeTranslator, "typeTranslator is null");
         requireNonNull(type, "type is null");
-        return new HiveType(typeTranslator.translate(type));
+        requireNonNull(defaultHiveType, "defaultHiveType is null");
+        return new HiveType(typeTranslator.translate(type, defaultHiveType));
+    }
+
+    public static Optional<HiveType> getDefaultHiveType(HiveStorageFormat storageFormat)
+    {
+        return storageFormat == PAGEFILE ? Optional.of(HIVE_BINARY) : Optional.empty();
```

Comment:
Let's do this only for temporary table

Reply:
It is only used for temporary table, I renamed this function to make it clear.